client/src/renderer/commands/shortcuts.js

```diff
@@ -1,71 +1,61 @@
 /** @format */
-import { genId } from "../helpers/uniqueIdGen.js";
+import { addShortcut } from "redux/actions";
+import * as fs from "fs";
 
 /**
- * Construct new shortcut object
- * @param {object} definition keyboard shortcut object (made with parseShortcut)
- * @param {string} definition string describing keyboard shortcut, to be parsed
- * @return {object} keyboard shortcut object
+ * get contents of JSON
+ *   open filestream
+ *   get file contents
+ *   put file contents into string
+ *   JSON.parse string into object
+ * search store for commands matching namespace:name
+ * for each namespace:name
+ *   get shortcut
+ *   pass shortcut to relevand command
  */
-export default function createShortcut(definition) {
-  if (typeof definition === "string") {
-    definition = parseShortcut(definition);
-  }
+export function readJSON() {
+  return new Promise((resolve, reject) => {
+    fs.readFile("./src/renderer/commands/keybinds.json", (err, data) => {
+      if (err) throw reject(err);
 
-  return {
-    id: genId(),
-    isAlt: false,
-    isCtrl: false,
-    isMeta: false,
-    isShift: false,
-    ...definition,
-  };
+      resolve(parseBindings(JSON.parse(data)));
+    });
+  });
 }
 
 /**
- * Parse string into shortcut object
- * @param {string} str string describing shortcut
- *                        Must be of the form "modifier[+modifier]+key"
- *                        No spaces, key at end, modifiers in any order
- *                        Must not be empty
- * @return {object} keyboard shortcut object
- *
- * @example
- *     parseShortcut("control+c")
- *     will return {
- *       "isAlt": false,
- *       "isCtrl": true,
- *       "isMeta": false,
- *       "isShift": false,
- *       "key": "c",
- *     }
+ * Parse keybinds into an array of shortcuts
+ * @param  {JSON} keybinds keybinds, parsed from keybinds.JSON
+ * @return {Array}          shortcuts, extracted from keybinds object
  */
-export function parseShortcut(str) {
-  if (str === "") throw "Cannot create an empty shortcut";
-
-  const modifiers = str.toLowerCase().split("+");
-
-  return {
-    isAlt: modifiers.includes("alt"),
-    isCtrl: modifiers.includes("control"),
-    isMeta: modifiers.includes("meta"),
-    isShift: modifiers.includes("shift"),
-    key: modifiers[modifiers.length - 1],
+export function parseBindings(keybinds) {
+  const parsed = {};
+  const addBinding = (id, binding) => {
+    parsed[id] = binding;
   };
+
+  parseBindingsRecursively(keybinds, addBinding);
+  return parsed;
 }
 
 /**
- * Compare registered shortcut with keydown event
- * @param  {shortcut}       shortcut registered shortcut object
- * @param  {event} event    caught keydown event
- * @return {boolean}        true if keydown event matches shortcut description
+ * Recursively traverse keybinds, extracting shortcut objects
+ * @param  {Object} obj        parsed JSON
+ * @param  {callback} addBinding callback function, adds binding to list of shortcuts
```

Comment:
You can define function arguments if you like. 
```suggestion
 * @param {function(id: string, binding: Object)} addBinding callback function,
 * adds binding to list of shortcuts.
```